### ISSUE TITLE
Adds new lsp for ruff

### ISSUE THREAD

**Issue by guyzmo** [mason-org/mason-registry#7203]
2 months ago was released `ruff server` a new LSP integration [replacing `ruff lsp`](https://github.com/astral-sh/ruff-lsp) already included with ruff.

The new implementation has [breaking changes](https://docs.astral.sh/ruff/editors/migration/) (the configuration options are not the same, so I guess it should be a new third option compared to `ruff-lsp` and `ruff` that could be called `ruff-server`.

I've considered making a PR to add this, but I got one question, how can I indicate that the implementation's binary should be called through a parameter: `ruff server`?

**Comment by mehalter** [comment]
This is already supported by just installing `ruff` from Mason

**Comment by Conarius** [comment]
Yeah, it just would need some adjustment in neovim/nvim-lspconfig to support that.

**Comment by mehalter** [comment]
As someone who uses the `ruff` package in Mason for LSP support, it doesn't require any modifications :)

**Comment by chrisgrieser** [comment]
Can also confirm, the existing ruff package already includes the LSP.
